homeassistant/components/upb/services.yaml

```diff
@@ -1,88 +1,161 @@
 light_fade_start:
+  name: Start light fade
   description: Start fading a light either up or down from current brightness.
+  target:
+    entity:
+      integration: upb
+      domain: light
   fields:
-    entity_id:
-      description: Name(s) of lights to start fading
-      example: "light.kitchen"
     brightness:
-      description: Number between 0 and 255 indicating brightness, where 0 turns the light off, 1 is the minimum brightness and 255 is the maximum brightness.
-      example: 142
+      name: Brightness
+      description: Number indicating brightness, where 0 turns the light off, 1 is the minimum brightness and 255 is the maximum brightness.
+      selector:
+        number:
+          min: 0
+          max: 255
     brightness_pct:
-      description: Number between 0 and 100 indicating percentage of full brightness, where 0 turns the light off, 1 is the minimum brightness and 100 is the maximum brightness.
-      example: 42
+      name: Brightness percentage
+      description: Number indicating percentage of full brightness, where 0 turns the light off, 1 is the minimum brightness and 100 is the maximum brightness.
+      selector:
+        number:
+          min: 0
+          max: 100
+          unit_of_measurement: '%'
     rate:
+      name: Rate
       description: Rate for light to transition to new brightness
-      example: 3
+      default: -1
+      selector:
+        number:
+          min: -1
+          max: 3600
+          step: 0.01
+          unit_of_measurement: seconds
 
 light_fade_stop:
+  name: Stop light fade
   description: Stop a light fade.
-  fields:
-    entity_id:
-      description: Name(s) of lights to stop fadding
-      example: "light.kitchen, light.family_room"
+  target:
+    entity:
+      integration: upb
+      domain: light
 
 light_blink:
+  name: Blink light
   description: Blink a light
+  target:
+    entity:
+      integration: upb
+      domain: light
   fields:
-    entity_id:
-      description: Name(s) of lights to start fading
-      example: "light.kitchen"
     rate:
-      description: Number of seconds between 0 and 4.25 that the link flashes on.
-      example: 4.2
+      name: Rate
+      description: Amount of time that the link flashes on.
+      default: 0.5
+      selector:
+        number:
+          min: 0
+          max: 4.25
+          step: 0.01
+          unit_of_measurement: seconds
 
 link_deactivate:
+  name: Deactivate link
   description: Deactivate a UPB scene.
-  fields:
-    entity_id:
-      description: Name(s) of scenes to deactivate
-      example: "scene.hygge"
+  target:
+    entity:
+      integration: upb
+      domain: light
 
 link_goto:
+  name: Go to link
   description: Set scene to brightness.
+  target:
+    entity:
+      integration: upb
+      domain: light
```

Comment:
The example was a scene, this domain is light.

Reply:
This comment applies to all UPB selectors.

Reply:
```suggestion
      domain: scene
```